src/main/kotlin/no/skatteetaten/aurora/boober/utils/CollectionUtils.kt

```diff
@@ -43,4 +47,21 @@ fun <T> Collection<T>?.nullOnEmpty(): Collection<T>? {
     }
 
     return this
+}
+
+// function that filters out everything except the lines with the provided keys from a .properties ByteArray
+fun filterProperties(properties: ByteArray, keys: List<String>) : ByteArray {
+    val props = Properties()
+    props.load(ByteArrayInputStream(properties))
+
+    val filteredProps = Properties()
+    keys.forEach( { key ->
+        if (props.stringPropertyNames().contains(key)) {
+            filteredProps.put(key, props.getProperty(key))
+        }
+    })
+
+    val baos = ByteArrayOutputStream()
```

Comment:
Her kan man også bruke .let - en smakssak, dog.

Reply:
God smak egentlig, bare uvant for en Kotlin-novise

Reply:
hva synes vi om lesbarheten på slikt (nested let i forhold til å deklarere et par vals)?

```
fun filterProperties(properties: ByteArray, keys: List<String>) : ByteArray {
    Properties().let { props ->
        props.load(ByteArrayInputStream(properties))

        Properties().let { newProps ->
            keys.forEach( { key ->
                if (props.stringPropertyNames().contains(key)) {
                    newProps.put(key, props.getProperty(key))
                }
            })
            ByteArrayOutputStream().let { baos ->
                newProps.store(baos, "Properties filtered.")
                return baos.toByteArray()
            }
        }
    }
}
```

Reply:
I utgangspunktet liker jeg flat kode.

Reply:
hva med 
val props=Properties().let{` it.load(...)}
val filtered=props.filter(keys)
return filtered.toByteArray()

og lage extension methods på properties for filtered og toByteArray?

Reply:
extension methods! 
Kotlin skill +1